### ISSUE TITLE
[BUG] plot_series inconsistent date/time index 

### ISSUE THREAD

**Issue by ngupta23** [alan-turing-institute/sktime#519]
**Describe the bug**

**To Reproduce**

```python
y = load_airline()
y = y[:48]
fh=np.arange(1, 13)

y_train, y_test = temporal_train_test_split(y, test_size=len(fh))
plot_series(y_train, y_test, labels=["y_train", "y_test"]);
print(y.shape, y_train.shape[0], y_test.shape[0])
print(y.index)
```

> (48,) 36 12
> PeriodIndex(['1949-01', '1949-02', '1949-03', '1949-04', '1949-05', '1949-06',
>              '1949-07', '1949-08', '1949-09', '1949-10', '1949-11', '1949-12',
>              '1950-01', '1950-02', '1950-03', '1950-04', '1950-05', '1950-06',
>              '1950-07', '1950-08', '1950-09', '1950-10', '1950-11', '1950-12',
>              '1951-01', '1951-02', '1951-03', '1951-04', '1951-05', '1951-06',
>              '1951-07', '1951-08', '1951-09', '1951-10', '1951-11', '1951-12',
>              '1952-01', '1952-02', '1952-03', '1952-04', '1952-05', '1952-06',
>              '1952-07', '1952-08', '1952-09', '1952-10', '1952-11', '1952-12'],
>             dtype='period[M]', name='Period', freq='M')
> 

**Expected behavior**
The X-axis should go till end of 1952. The X-axis on the plot does not reflect the indices correctly. It should go till 1952, but shows values only till 1949

![image](https://user-images.githubusercontent.com/33585645/100865855-800ef980-345d-11eb-9980-476454e8de68.png)

**Versions**
System:
    python: 3.6.12 |Anaconda, Inc.| (default, Sep  9 2020, 00:29:25) [MSC v.1916 64 bit (AMD64)]
executable: C:\Users\xxxx\AppData\Local\Continuum\anaconda3\envs\sktime\python.exe
   machine: Windows-10-10.0.18362-SP0

Python dependencies:
          pip: 20.3
   setuptools: 49.6.0
      sklearn: 0.23.2
        numpy: 1.19.2
        scipy: 1.5.2
       Cython: 0.29.17
       pandas: 1.1.3
   matplotlib: 3.3.2
       joblib: 0.17.0
        numba: None
     pmdarima: 1.7.1
      tsfresh: None

**Comment by mloning** [comment]
Thanks for raising the issue @ngupta23 - looks like internally something goes wrong when merging the indices from `y_train` and `y_test`.

**Comment by gracewgao** [comment]
I can try to fix this!

**Comment by mloning** [comment]
@gracewgao that would be awesome! I noticed that integer indices also seem a bit off. Let me know if you need any help.

**Comment by wolfiex** [comment]
I have a similar issue appearing in 0.5.2. 

Python  3.8 
Mac OS
```
import sktime
sktime.__version__

# returns '0.5.2'
```
![Screen Shot 2021-01-14 at 18 22 37](https://user-images.githubusercontent.com/15582577/104715137-089bdd80-571e-11eb-944c-60304a2e0755.png)

**Comment by mloning** [comment]
thanks @wolfiex for posting it here

**Comment by mloning** [comment]
If you're interested in looking into this, I'd appreciate a PR!

**Comment by wolfiex** [comment]
@mloning  

The issue seems likely to a problem with the building of v 0.5.2. Inserting the `plotting.py` code directly inline solve the problem.  I assume that there was an issue when building the latest version. My install uses the 'conda-forge' channel on anaconda. 

The first is imported as `from sktime.utils.plotting import plot_series` whereas the second (plot_series_github) is a copy of the plotting.py file within the utils folder. 

```
conda_forge = plot_series(y_train, y_test, labels=["y_train", "y_test"])
inline_code = plot_series_github(y_train, y_test, labels=["y_train", "y_test"])
```

![Screen Shot 2021-01-15 at 16 11 18](https://user-images.githubusercontent.com/15582577/104750824-824ac000-574c-11eb-8412-ca987694aed6.png)

**Comment by wolfiex** [comment]
Having just checked the files, apart from two lines of whitespace (likely my editor) the two files are identical! It seems that something is going wrong when importing the function, but not when pasted inline.

**Comment by mloning** [comment]
@wolfiex very strange, so only the conda version is broken?

**Comment by wolfiex** [comment]
> @wolfiex very strange, so only the conda version is broken?

The conda install. The file itself seems identical.

**Comment by mloning** [comment]
@wolfiex could you try to create a minimal, reproducible example? 

This seems to work correctly for me: 
```python
from sktime.datasets import load_airline
from sktime.forecasting.model_selection import temporal_train_test_split
from sktime.utils.plotting import plot_series
import numpy as np

y = load_airline()
fh = np.arange(1, 13)

y_train, y_test = temporal_train_test_split(y, test_size=len(fh))
fig, ax = plot_series(y_train, y_test, labels=["y_train", "y_test"]);
print(y.index)
```

**Comment by jpss95** [comment]
Hi all, the same thing is happening to me with the plot_windows method.

Here a code to reproduce the bug:

import pandas as pd
import numpy as np
from sktime.utils.plotting import plot_windows
from sktime.forecasting.base import ForecastingHorizon
from sktime.forecasting.model_selection import ExpandingWindowSplitter

forecast_steps= 5
initial_window_size= 40
step_length=1
fh = ForecastingHorizon(np.arange(1,forecast_steps+1))

#create dummy data
dates = pd.date_range(start='2016-01', end='2022-12', freq='M')
df = pd.DataFrame(index=dates)
df['y'] = np.random.rand(len(df))

cv = ExpandingWindowSplitter(fh=fh, initial_window=initial_window_size, step_length=step_length)

print(df)
plot_windows(cv, df)


![image](https://github.com/sktime/sktime/assets/64998671/b4bc55db-e44f-4d1c-b1b4-f5251455e7bb)


The image clearly has wrong indexes

cc: @fkiraly
